src/main/java/ru/ayaz/ChatServer.java

```diff
@@ -0,0 +1,163 @@
+package ru.ayaz;
+
+
+import java.io.BufferedReader;
+import java.io.IOException;
+import java.io.InputStreamReader;
+import java.io.PrintWriter;
+import java.net.ServerSocket;
+import java.net.Socket;
+import java.util.HashMap;
+import java.util.Map;
+
+public class ChatServer {
+
+    private ServerSocket serverSocket;
+    private Map<String, User> usersMap;
+    private Map<String, PrintWriter> outputWriterMap;
+
+
+    public ChatServer() {
+        try {
+            this.serverSocket = new ServerSocket(4400);
+            this.usersMap = new HashMap<String, User>();
+            this.outputWriterMap = new HashMap<String, PrintWriter>();
+        } catch (IOException e) {
+            e.printStackTrace();
+        }
+    }
+
+
+    public void start() {
+
+        try {
+            while (true) {
+                Socket socket = serverSocket.accept();
+                Thread t = new Thread(new UserHandler(socket));
+                t.start();
+            }
+        } catch (IOException e) {
+            e.printStackTrace();
+        }
+    }
+
+
+    private void sendMessageToEveryone(String message, String sender) {
+
+        System.out.println(message);
+
+        for (Map.Entry<String, User> entry : usersMap.entrySet()) {
+            String nickname = entry.getKey();
+            User user = entry.getValue();
+            if (!nickname.equals(sender)) {
+                if (!user.isIgnored(sender)) {
+                    PrintWriter pw = outputWriterMap.get(nickname);
+                    pw.println(message);
+                    pw.flush();
+                }
+            }
+        }
+
+    }
+
+
+    public class UserHandler implements Runnable {
+
+        private User user;
+        private MessageProcessor processor;
+        private PrintWriter writer;
+        private BufferedReader reader;
+
+
+        public UserHandler(Socket socket) {
+            try {
+                this.reader = new BufferedReader(new InputStreamReader(socket.getInputStream()));
+                this.writer = new PrintWriter(socket.getOutputStream());
+                this.user = new User();
+                this.processor = new MessageProcessor();
+            } catch (IOException e) {
+                e.printStackTrace();
+            }
+        }
+
+
+        public void run() {
+
+            String message;
+            String sender;
+
+            welcomeNewUser();
+
+            try {
+
+                user.setNickname(listenForMessage());
+                usersMap.put(user.getNickname(), user);
```

Comment:
&#x1F34E; **Issue**
Better put this line in a separate method meaningfully named (e.g. "addUser")